hedgehog/src/Hedgehog/Internal/TH.hs

```diff
@@ -21,14 +21,38 @@ import           Language.Haskell.TH.Syntax
 type TExpQ a =
   Q (TExp a)
 
--- | Check all the properties in a file.
+-- | Check all the properties in a file sequentially.
+--
+-- > tests :: IO Bool
+-- > tests =
+-- >   $$(checkSequential)
+--
+checkSequential :: TExpQ (IO Bool)
```

Comment:
I should have mentioned this before, but I often wondered whether it would be better to generate just the bare minimum, and then do the "running" at the top level. So rather than `IO Bool`, just generate the bare minimum `Map Name Source`. That said, you may be planning to have files that are either sequential and some concurrent.